pmlc/rt/boundscheck.cc

```diff
@@ -0,0 +1,16 @@
+// Copyright 2020 Intel Corporation
+
+#include <stdint.h>
+
+#include <stdexcept>
+
+// This function is called from the BoundsCheckPass.
+// index - ndex for a dimension that is accessed by load/store operation
+// range - upper bound of the range for this dimension (it is always 0 -
+// range)
+extern "C" void plaidml_rt_bounds_check(intptr_t index, unsigned range) {
+  int64_t accessIndex = (int64_t)index;
+  int64_t upperRange = (int64_t)range;
+  if (accessIndex < 0 || accessIndex >= upperRange)
+    std::runtime_error("Out of bounds index for mlir::LoadOp or mlir::StoreOp");
```

Comment:
Does throwing actually work inside the MLIR JIT? Have you tested this?

Reply:
This is not in the Jit, I think. The code is already jitted. We are executing the Jitted code when the exception is thrown. I did test it by adding some code to the xsmm RT function and it did propagate the exception properly and failed the test.

Reply:
This will be executed by the JIT is what I mean.

Reply:
You mention that it "failed the test", can you add the test to this PR? I don't understand how you have verified that the exception properly gets thrown. How could it? At the time of this writing, it didn't even have a `throw`, so it was clearly broken.

Reply:
I talked to Jeremy about that. I vertifyed in the xsmm.cc, by adding tyo throw an exception and it worked as expected - exception propagated. So, that proved that mlir generated code beghaves properly in regard with exception handling.
This doesn't matter now, because I added the infra and test to allow exception case testing in the plmc jit.